bootenv.go

```diff
@@ -33,22 +34,39 @@ func NewEnvironment(cmd Commander) *uBootEnv {
 
 func (e *uBootEnv) ReadEnv(names ...string) (BootVars, error) {
 	getEnvCmd := e.Command("fw_printenv", names...)
-	return getOrSetEnvironmentVariable(getEnvCmd)
+	return getEnvironmentVariable(getEnvCmd)
 }
 
 func (e *uBootEnv) WriteEnv(vars BootVars) error {
-	//TODO: try to make this atomic later
+	// Make environment update atomic by using fw_setenv "-s" option.
+	setEnvCmd := e.Command("fw_setenv", "-s", "-")
+	pipe, err := setEnvCmd.StdinPipe()
```

Comment:
I recall @bboozzoo having some opinions about using pipes with command. I don't remember exactly what this was about. Maybe he can take a look as well.